### PR TITLE
Include `body` property in `HTTPError`

### PR DESCRIPTION
**Pull request by mrmlnc**
#### Description

The ability to look at the body of the error response inside the `retries` function .

#### Why?

Some services use the response body to provide an extended description of why the error occurred (for example, reason of [409 code](https://developer.mozilla.org/docs/Web/HTTP/Status/409)).

We would like to use this field for logging.

### CODE REVIEW

**Comment by mrmlnc** [comment]
Maybe it makes sense to also pass `request` as field of the error?

**Comment by szmarczak** [comment]
@sindresorhus Any thoughts?

File: source/errors.js [hunk 1]
```diff
@@ -59,7 +59,10 @@ module.exports.ParseError = class extends GotError {
 };
 
 module.exports.HTTPError = class extends GotError {
-	constructor(statusCode, statusMessage, headers, opts) {
+	constructor(request, opts) {
```
Comment (brandon93s) [inline source/errors.js]: Shouldn't parameter 1 be `response` instead of `request`?

Comment (szmarczak) [reply source/errors.js]: changed

**Comment by sindresorhus** [comment]
Makes sense.

The docs needs to be updated too: https://github.com/sindresorhus/got#gothttperror

**Comment by szmarczak** [comment]
> The docs needs to be updated too: [sindresorhus/got#gothttperror](https://github.com/sindresorhus/got#gothttperror)

done

**Comment by nuragic** [comment]
Hello! Any chance to bump a patch release for this one today? 😌 Thanks!

**Comment by razor-x** [comment]
@sindresorhus I am adopting Got for our project to replace `request`. So far it looks great, but I need this (and `timings`) in a new version before I can push stable stuff on my side (can't use the git version for internal reasons). 

Been using the git version to develop with no issues, but :smile_cat: Can haz has release  :hamburger:  or ETA plz? :pray:

**Comment by szmarczak** [comment]
@razor-x https://github.com/sindresorhus/got/milestone/7 also see #635

**Comment by Rowno** [comment]
Doesn't this introduce a security issue for people logging errors? Since the body could easily contain sensitive data.

**Comment by sindresorhus** [comment]
```js
class FooError extends Error {
	constructor() {
		super();
		this.unicorn = 'not logged';
    }
}

console.log(new FooError());
// Error
//  at <anonymous>:8:13
```

**Comment by Rowno** [comment]
I mean logging errors with packages like winston or sentry, which also log all the enumerable properties on the Error object.

**Pull request merged**